src/support/geocoder-results-parser.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace SearchApi\Support;
+
+
+use Exception;
+use SearchApi\Models as Models;
+
+/**
+ * Class Geo_Parser - Parses the returned result from a Geocoder
+ */
+class GeoParser {
+  public function reverse_geocoder_parser( $json_results ) {
```

Comment:
This function is doing too much, you can slim it up by separating out the creation of the $search_term_array into another function and have reverse_geocoder_parser call that. This will make testing easier too since you can test the logic for creating the $search_term_array separately from the logic that encodes and decodes the json response.